### ISSUE TITLE
Use of "npm run build" when node_env is set to development

### ISSUE THREAD

**Issue by lholmquist** [sclorg/s2i-nodejs-container#386]
### Version

All versions

### Description

While i was trying to add a client module(client is written in Typescript, so it has a "build step") to the test suite #384 , I noticed that when NODE_ENV is set to "development", the assemble script only does a `npm install`, but if the NODE_ENV is set to production,  the assemble script will also run `npm run build --if-present`.   

https://github.com/sclorg/s2i-nodejs-container/blob/master/18/s2i/bin/assemble#LL81-L93C28
```
if [ "$NODE_ENV" != "production" ]; then

	echo "---> Building your Node application from source"
	npm install

else

	echo "---> Installing all dependencies"
	NODE_ENV=development npm install

	#do not fail when there is no build script
	echo "---> Building in production mode"
	npm run build --if-present

......
```

Since the assemble script, doesn't do the build step, the client module tests fail since those tests are looking for a "compiled" file.

I can get around this easily enough, by putting the build step in a pretest npm script.

The question i had was if there was a reason why `npm run build --if-present` was left out in NODE_ENV isn't production?

**Comment by pkubatrh** [comment]
The split between development and prod was introuced in https://github.com/sclorg/s2i-nodejs-container/pull/152 and I guess only `npm install` was left in there to make sure we have a way to go back to the exact same behavior as before.
Even back then there was a suggestion to add the `npm run build` steps (https://github.com/sclorg/s2i-nodejs-container/pull/152#discussion_r164525441) but the owner of the PR decided against.

In other words - I found no strong reasons it is not called in production builds. Feel free to follow up with a PR that adds the call if it seems like the better approach.

**Comment by hhorak** [comment]
I agree with Petr here, although I have to admit I'm not that familiar with the specifics of npm evnironments, so an opinion of somebody who is more familiar with Node.js environment generally would be appreciated.

**Comment by lholmquist** [comment]
going to close this since the work around is probably fine to use
